feature: support for DMS in position control

steff-o opened the pull request:
Closes #1602. This PR contains code to format coordinates in the position control  in DMS-format and also enter coordinates in DMS-format in the go to coordinate function.

As the existing configuration made it impossible to add the same coordinate system twice, I added a new configuration format. The old one is still valid for backwards compatibility, but lacks the new functionality.

In the new configuration format it is possible to specify dms as format and also an optional precision parameter that controls how many decimals a coordinate should have.

Example configuration:
```json
{
      "name": "position",
      "options": {
        "title": "Web Mercator",
        "projections": [
          {
            "projectionCode": "EPSG:4326",
            "projectionLabel": "Wgs84"
          },
          {
            "projectionCode": "EPSG:4326",
            "projectionLabel": "Wgs84 DMS",
            "dms": true,
            "precision": 2
          }
          ]
      }
    }
```

The big difference is that `projections` now is an array instead of an object in order to add the same coordinate system several times with different format.

The `projectionCode` property is mandatory and corresponds to the old configuration format's object key

The `projectionLabel` property is mandatory and specifies the text on the coordinate system selector button. It corresponds to the old configuration format's object value.

The `dms` property is optional and if set to true it is formatted as DMS.

The `precision` property is optional and if specified it sets the number of decimals. It works for both DMS and non-DMS. If not specified it defaults to 5 for EPSG:4326 in nonDMS format and 0 for everything else (same as before)

johnnyblasta commented:
I'm trying to change position in DMS, but are getting an alert that I'm outside the map extent although it shouldn't be outside.

steff-o commented:
Strange. Never changed that stuff. It seems to check against both map extent and coordinate system extent, so if you have a local coordinate system (like sweref 17 15) it may throw on that.

Other possibility is that I messed up in transforming coordinates to map projection or parsing is just plain wrong.

I will look in to it!

steff-o commented:
My bad, I simplified the calculation of seconds * 1/3600 and kept the * instead of / after lint complained and obviously didn't test after. Fixed now.

johnnyblasta commented on src/controls/position.js:
```diff
@@ -112,47 +137,58 @@ const Position = function Position(options = {}) {
-  }
-
+  /**
+   * Update coords in input, transformin if necessary.
```
Missing g i transforming

johnnyblasta reviewed (approved):
LGTM, except some misspellings in comments

steff-o commented:
Tough crowd... Fixed comments.
